test/object-store/util/sync/redirect_server.hpp

```diff
@@ -0,0 +1,282 @@
+        })
+    {
+        m_acceptor.open(m_endpoint.protocol());
+        m_acceptor.bind(m_endpoint);
+        m_endpoint = m_acceptor.local_endpoint();
+        m_acceptor.listen();
+        m_service.post([this](Status status) {
+            REALM_ASSERT(status.is_ok());
+            do_accept();
+        });
+    }
+
+    ~RedirectingHttpServer()
+    {
+        m_acceptor.cancel();
+        m_service.stop();
+        m_server_thread.join();
+    }
+
+    std::string base_url() const
+    {
+        return util::format("http://localhost:%1", m_endpoint.port());
+    }
+
+private:
+    struct BufferedSocket : network::Socket {
+        BufferedSocket(network::Service& service)
+            : network::Socket(service)
+        {
+        }
+
+        BufferedSocket(network::Service& service, const network::StreamProtocol& protocol,
+                       native_handle_type native_handle)
+            : network::Socket(service, protocol, native_handle)
+        {
+        }
+
+
+        template <class H>
+        void async_read_until(char* buffer, std::size_t size, char delim, H handler)
+        {
+            network::Socket::async_read_until(buffer, size, delim, m_read_buffer, std::move(handler));
+        }
+
+        template <class H>
+        void async_read(char* buffer, std::size_t size, H handler)
+        {
+            network::Socket::async_read(buffer, size, m_read_buffer, std::move(handler));
+        }
+
+    private:
+        network::ReadAheadBuffer m_read_buffer;
+    };
+
+    struct Conn : public util::RefCountBase, websocket::Config {
+        Conn(network::Service& service, const std::shared_ptr<util::Logger>& logger)
+            : random(Catch::getSeed())
+            , logger(logger)
+            , socket(service)
+            , http_server(socket, logger)
+        {
+        }
+
+        // Implement the websocket::Config interface
+        const std::shared_ptr<util::Logger>& websocket_get_logger() noexcept override
+        {
+            return logger;
+        }
+
+        std::mt19937_64& websocket_get_random() noexcept override
+        {
+            return random;
+        }
+
+        void async_write(const char* data, size_t size, websocket::WriteCompletionHandler handler) override
+        {
+            socket.async_write(data, size, std::move(handler));
+        }
+
+        void async_read(char* buffer, size_t size, websocket::ReadCompletionHandler handler) override
+        {
+            socket.async_read(buffer, size, std::move(handler));
+        }
+
+        void async_read_until(char* buffer, size_t size, char delim,
+                              websocket::ReadCompletionHandler handler) override
+        {
+            socket.async_read_until(buffer, size, delim, std::move(handler));
+        }
+
+        void websocket_handshake_completion_handler(const HTTPHeaders&) override {}
+
+        void websocket_read_error_handler(std::error_code) override {}
```

Comment:
Should any of these default handlers at least log the error? Or is it not expected to ever be called - would a REALM_UNREACHABLE() be appropriate?